### PR TITLE
UI: Ember 5 deprecations: filterBy, mapBy, auto location

### PR DESCRIPTION
**Pull request by hashishaw**
This PR updates the usage of filterBy and mapBy which were are both deprecated for EmberData 5.0 (no docs for it, but the error message is below)

```
The mapBy method on the class RecordArray is deprecated. Use the native array method map instead. [deprecation id: ember-data:deprecate-array-like] This will be removed in ember-data 5.0.
```

`findBy` is still valid for Mirage, so we do have a few more instances when you find within the UI directory. 

This PR also removes the only known reference to `auto` location which is [also deprecated](https://deprecations.emberjs.com/v4.x/#toc_deprecate-auto-location)

- [x] Enterprise tests pass locally

### CODE REVIEW

File: ui/app/adapters/secret-engine.js [hunk 1]
```diff
@@ -132,7 +132,11 @@ export default ApplicationAdapter.extend({
 
   saveZeroAddressConfig(store, type, snapshot) {
     const path = encodePath(snapshot.id);
-    const roles = store.peekAll('role-ssh').filterBy('zeroAddress').mapBy('id').join(',');
+    const roles = store
+      .peekAll('role-ssh')
+      .filter((role) => role.zeroAddress === true)
```
Comment (kiannaquach) [inline ui/app/adapters/secret-engine.js:137]:
```suggestion
      .filter((role) => role.zeroAddress)
```
I think should work. I tried this in the console:
```
var arr = [{ selected: true }, { selected: true }, { selected: false }]
arr.filter(a => a.selected) // => returns [{ selected: true }, { selected: true }]
```

File: ui/app/components/wizard/features-selection.js [hunk 2]
```diff
@@ -134,7 +134,7 @@ export default Component.extend({
   showReplication: or('version.hasPerfReplication', 'version.hasDRReplication'),
 
   selectedFeatures: computed('allFeatures.@each.selected', function () {
-    return this.allFeatures.filterBy('selected').mapBy('key');
+    return this.allFeatures.filter((feature) => feature.selected === true).map((feature) => feature.key);
```
Comment (kiannaquach) [inline ui/app/components/wizard/features-selection.js:137]:
```suggestion
    return this.allFeatures.filter((feature) => feature.selected).map((feature) => feature.key);
```

**Review by kiannaquach — APPROVED** [review]
thanks for tackling this, chelsea! i only had one minor suggestion but everything looks good!

File: ui/app/adapters/secret-engine.js (hunk 1 above)
Comment (hashishaw) [reply ui/app/adapters/secret-engine.js:137]: I was being overly cautious about making sure it's a boolean truthy value, but I think this should work

**Pull request merged**